components/boards/AllArticlesSection.tsx

```diff
@@ -0,0 +1,160 @@
+import SearchBar from "@/components/ui/SearchBar";
+import DropdownMenu from "@/components/ui/DropdownMenu";
+import { useEffect, useState } from "react";
+import LikeCountDisplay from "@/components/ui/LikeCountDisplay";
+import EmptyState from "@/components/ui/EmptyState";
+import { useRouter } from "next/router";
+
+const ItemContainer = styled(Link)``;
+
+const ArticleInfoDiv = styled(FlexRowCentered)`
+  gap: 8px;
+  color: var(--gray-600);
+  font-size: 14px;
+`;
+
+interface ArticleItemProps {
+  article: Article;
+}
+
+const ArticleItem: React.FC<ArticleItemProps> = ({ article }) => {
+  const dateString = format(article.createdAt, "yyyy. MM. dd");
+
+  return (
+    <>
+      <ItemContainer href={`/boards/${article.id}`}>
+        <MainContent>
+          <ArticleTitle>{article.title}</ArticleTitle>
+          {article.image && (
+            <ArticleThumbnail>
+              {/* Next Image의 width, height을 설정해줄 것이 아니라면 부모 div 내에서 fill, objectFit 설정으로 비율 유지하면서 유연하게 크기 조정 */}
+              {/* 프로젝트 내에 있는 이미지 파일을 사용하는 게 아니라면 next.config.mjs에 이미지 주소 설정 필요 */}
+              <ImageWrapper>
+                <Image
+                  fill
+                  src={article.image}
+                  alt={`${article.id}번 게시글 이미지`}
+                  style={{ objectFit: "contain" }}
+                />
+              </ImageWrapper>
+            </ArticleThumbnail>
+          )}
+        </MainContent>
+
+        <ArticleInfo>
+          <ArticleInfoDiv>
+            {/* ProfilePlaceholder 아이콘의 SVG 파일에서 고정된 width, height을 삭제했어요 */}
+            {/* <ProfilePlaceholder width={24} height={24} /> */}
+            {article.writer.nickname} <Timestamp>{dateString}</Timestamp>
+          </ArticleInfoDiv>
+
+          <LikeCountDisplay count={article.likeCount} iconWidth={24} gap={8} />
+        </ArticleInfo>
+      </ItemContainer>
+
+      <LineDivider $margin="24px 0" />
+    </>
+  );
+};
+
+const AddArticleLink = styled(StyledLink)``;
+
+interface AllArticlesSectionProps {
+  initialArticles: Article[];
+}
+
+const AllArticlesSection: React.FC<AllArticlesSectionProps> = ({
+  initialArticles,
+}) => {
+  const [orderBy, setOrderBy] = useState<ArticleSortOption>("recent");
+  const [articles, setArticles] = useState(initialArticles);
+
+  const router = useRouter();
+  const keyword = (router.query.q as string) || "";
+
+  const handleSortSelection = (sortOption: ArticleSortOption) => {
+    setOrderBy(sortOption);
+  };
+
+  const handleSearch = (searchKeyword: string) => {
+    const query = { ...router.query };
+    if (searchKeyword.trim()) {
+      query.q = searchKeyword;
+    } else {
+      delete query.q; // Optional: 키워드가 빈 문자열일 때 URL에서 query string 없애주기
+    }
+    router.replace({
+      pathname: router.pathname,
+      query,
+    });
+  };
+
+  useEffect(() => {
+    const fetchArticles = async () => {
+      let url = `https://panda-market-api.vercel.app/articles?orderBy=${orderBy}`;
```

Comment:
P3:
공통으로 사용되는 기본 주소는 변수에 저장해두고 쓰셔도 좋을 것 같아요~
```suggestion
      let url = `${BASE_URL}/articles?orderBy=${orderBy}`;
```